Mathlib/LinearAlgebra/Matrix/Permanent.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2024 Moritz Firsching. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching
+-/
+import Mathlib.Data.Fintype.Perm
+import Mathlib.Data.Matrix.Basic
+/-!
+# permanentanent of a matrix
+
+This file defines the permanentanent of a matrix, `Matrix.permanent`, and some of its properties.
+
+## Main definitions
+
+* `Matrix.permanent`: the permanentanent of a square matrix, as a sum over permanentutations
+
+-/
+
+open Equiv Fintype Finset
+
+namespace Matrix
+
+open Matrix
```

Comment:
You don't need to open namespaces when you're already inside them. (Sorry, I missed this earlier.)